### PR TITLE
[#10356] Instructor view results: multiple photo popups

### LINKED ISSUE #10356: Instructor view results: multiple photo popups
image

These things pops up (and stays) as you move the move across the page

### PR DESCRIPTION
**Pull request by t-cheepeng**
Fixes #10356 

**Outline of Solution**
- Add icon to start of name that user have to click to get popover
- Change behaviour to always require a click on icon to view photo
- Rename variable to be better reflect code

t

### CODE REVIEW

**Comment by wkurniawan07** [comment]
I think we need a tooltip or something to tell that clicking that icon will allow the photo to appear. @ccyccyccy what do you think?

**Comment by damithc** [comment]
> I think we need a tooltip or something to tell that clicking that icon will allow the photo to appear. @ccyccyccy what do you think?

I agree. One option is to give an unobtrusive (e.g., small text, grey color) info message at the top of the table/page like this.

<sub>(i) Click on the X icon to view ...</sub>

Team | Student | ...
------|----------|---
...|....|...

**Comment by ccyccyccy** [comment]
> I think we need a tooltip or something to tell that clicking that icon will allow the photo to appear. @ccyccyccy what do you think?

Either tooltip or an info message should work well enough I think. For this type of quite unused feature, I would prefer tooltip over an info message if we cannot fit the info message in well enough.

Also, I think the icon should darken or give some kind of reaction instantly when hovered over.

**Comment by t-cheepeng** [comment]
@wkurniawan07 @ccyccyccy @damithc Do give the screencap another look. Updated the behaviour to make icon fill up on hover and a tooltip to appear

**Comment by damithc** [comment]
> @wkurniawan07 @ccyccyccy @damithc Do give the screencap another look. Updated the behaviour to make icon fill up on hover and a tooltip to appear

Looks good @t-cheepeng

File: src/web/app/components/teammates-common/student-name/student-name-with-photo.component.ts [hunk 1]
```diff
@@ -19,6 +19,8 @@ export class StudentNameWithPhotoComponent implements OnInit {
   @Input()
   email: string = '';
 
+  iconClass: string = 'far';
```
Comment (ccyccyccy) [inline src/web/app/components/teammates-common/student-name/student-name-with-photo.component.ts:22]: Why not use `:hover` instead?

**Comment by t-cheepeng** [comment]
t
Capture
d

@ccyccyccy Removed the spaces

**Comment by t-cheepeng** [comment]
@ccyccyccy  Updated screenshots above. Thanks for double checking! 😄

**Comment by ccyccyccy** [comment]
After a closer look, maybe there's a simpler method, where `showPhotoEvent` could call `t.close()`directly, eliminating the need for the more complex timing code? Not sure if that would work.

**Comment by t-cheepeng** [comment]
> `showPhotoEvent` could call `t.close()`

Ah yes, I did try this method out. But i couldn't get it to work. If we did not have our own timeout, the default timeout from `ngbTooltipConfig` will happen and we have no way of clearing the internal timeout(or however it is implemented) since it is abstracted away by the config. If we used our own timeout, just calling `t.close()` in the `showPhotoEvent` would not work as the event would fire before the timeout completes so the tooltip will still appear after timeout completes(which was the problem where you posted in the screenshot above; clicking on the icon before tooltip appears)

**Comment by ccyccyccy** [comment]
> Ah yes, I did try this method out. But i couldn't get it to work. If we did not have our own timeout, the default timeout from `ngbTooltipConfig` will happen and we have no way of clearing the internal timeout(or however it is implemented) since it is abstracted away by the config. If we used our own timeout, just calling `t.close()` in the `showPhotoEvent` would not work as the event would fire before the timeout completes so the tooltip will still appear after timeout completes(which was the problem you posted in the screenshot above; clicking on the icon before tooltip appears)

Maybe can listen to the `shown` event from the tooltip?

**Comment by madanalogy** [comment]
Just change the status back to `s.ToMerge` when the discussion is resolved

**Comment by t-cheepeng** [comment]
> Maybe can listen to the shown event from the tooltip? Then call close() instantly

@ccyccyccy This would work but we would have to decide in the callback if the tooltip should be closed(close if photo popover is shown, don't close if popover is not shown) which would then require parent component to have some sort of `ViewChild` or control over the state of the popover. In the end, the code looks equally complex so I see no real benefit to switching to this

**Pull request merged**